geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/SQLSpatialAccessorFunctions.scala

```diff
@@ -74,6 +76,26 @@ object SQLSpatialAccessorFunctions {
     case _ => null
   }
 
+  implicit def st_boundary = udf(ST_Boundary)
+  implicit def st_coordDim = udf(ST_CoordDim)
+  implicit def st_dimension = udf(ST_Dimension)
+  implicit def st_envelope = udf(ST_Envelope)
+  implicit def st_exteriorRing = udf(ST_ExteriorRing)
+  implicit def st_geometryN = udf(ST_GeometryN)
+  implicit def st_geometryType = udf(ST_GeometryType)
+  implicit def st_interiorRingN = udf(ST_InteriorRingN)
+  implicit def st_isClosed = udf(ST_IsClosed)
+  implicit def st_isCollection = udf(ST_IsCollection)
+  implicit def st_isEmpty = udf(ST_IsEmpty)
+  implicit def st_isRing = udf(ST_IsRing)
+  implicit def st_isSimple = udf(ST_IsSimple)
+  implicit def st_isValid = udf(ST_IsValid)
+  implicit def st_numGeometries = udf(ST_NumGeometries)
+  implicit def st_numPoints = udf(ST_NumPoints)
+  implicit def st_pointN = udf(ST_PointN)
+  implicit def st_x = udf(ST_X)
+  implicit def st_y = udf(ST_Y)
```

Comment:
If the purpose of the above is to make them more readily available to DataFrame users, I highly recommend using the following approach to promote more readable schemas and execution plans:

https://github.com/s22s/raster-frames/blob/54b8b74f6b1971ed0bbbe3ba5696c932d8f8d3fa/src/main/scala/astraea/spark/rasterframes/jts/SpatialPredicates.scala#L41-L61

It also allows you to return them as `TypedColumn[Any, T]`, more readily allowing for lifting into `Dataset[T]` space.

Reply:
@metasim Got it. Looking at your example I noticed you expose some UDFs both as functions of their columns and as functions of their literal values.
https://github.com/s22s/raster-frames/blob/54b8b74f6b1971ed0bbbe3ba5696c932d8f8d3fa/src/main/scala/astraea/spark/rasterframes/jts/SpatialConverters.scala#L57-L62
Would it be useful to expose all functions as pairs like that?

Reply:
@atallahhezbor I /think/ so? I feel like it reduces friction for the user when using the Dataset API. I'm only hesitant to strongly suggest it because it's a bit new in RasterFrames and I haven't fully convinced myself that it won't evolve further.